python_modules/dagster/dagster/_core/definitions/decorators/asset_decorator.py

```diff
@@ -549,6 +549,12 @@ def create_assets_def_from_fn_and_decorator_args(
     emit_runtime_warning=False,
     breaking_version="1.10.0",
 )
+@hidden_param(
+    param="allow_arbitrary_check_specs",
+    emit_runtime_warning=False,
+    # does this actually need to be set?
```

Comment:
hidden_param was originally intended for deprecating things rather than introducing new hidden things, but both feel like valid uses. kinda weird to set this to any value (including the empty string), maybe would be good to extend the hidden_param "framework" to have a "hidden_internal_param" decorator that doesn't take a breaking_version argument and never emits a runtime warning?